### ISSUE TITLE
aws_elasticache_cluster cluster_id is rejected when an existing primary node has a 22-character ID

### ISSUE THREAD

**Issue by lavaturtle** [terraform-providers/terraform-provider-aws#5381]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.11.7
+ provider.aws v1.29.0
+ provider.template v1.0.0

### Affected Resource(s)

* aws_elasticache_cluster

### Terraform Configuration Files

```hcl
resource "aws_elasticache_cluster" "redis" {
  cluster_id = "${var.redis_cluster_name}-001"
  engine = "redis"
  node_type = "${var.redis_node_type}"
  snapshot_retention_limit = "${var.redis_snapshot_retention_limit}"
...
}
```
The `redis_cluster_name` variable is an 18-character string, like "my-kinda-long-name"

I have a lot of other config, but it is not relevant to this issue.

### Expected Behavior

I have an existing cluster in my state file whose primary node ID is "my-kinda-long-name-001". I expected terraform to show no change to its ID.

### Actual Behavior

I get an error message: `aws_elasticache_cluster.redis: "cluster_id" ("my-kinda-long-name-001") must contain from 1 to 20 alphanumeric characters or hyphens`

### Steps to Reproduce

1. Using terraform, create an elasticache redis cluster with a shorter name. Notice that the `cluster_id` is the ID of the primary node in the cluster.
2. Using the web console, create an elasticache redis cluster named "my-kinda-long-name". Its primary node should be named "my-kinda-long-name-001".
3. Edit the state file to replace the `cluster_id` of the terraform-managed cluster with `my-kinda-long-name-001`.
4. `terraform apply`
Expected: Terraform shows any diff between the `my-kinda-long-name` cluster and the config.
Actual: Error about cluster_id length.

It seems like AWS only applies the 20-character limit to the cluster itself, but terraform applies it to the primary node ID, causing it to reject as invalid some existing node IDs.

**Comment by bflad** [comment]
The Elasticache service now supports cluster identifiers up to 50 characters. We will support this new limit via: https://github.com/terraform-providers/terraform-provider-aws/pull/9941

**Comment by bflad** [comment]
The above has been merged and will release in version 2.28.0 of the Terraform AWS Provider, Thursday. 👍
